git/refs/symbolic.py

```diff
@@ -313,7 +313,7 @@ def set_reference(self, ref, logmsg=None):
 
         lfd = LockedFD(fpath)
         fd = lfd.open(write=True, stream=True)
-        fd.write(write_value.encode('ascii'))
+        fd.write(write_value.encode('ascii') + '\n')
```

Comment:
For Python 3 compat, prefer to use `b'\n'` here, since this should be a byte, not a unicode string.